akka-sample-sharding-scala/src/main/scala/sample/sharding/ShardingApp.scala

```diff
@@ -15,16 +20,19 @@ object ShardingApp {
     // In a production application you wouldn't typically start multiple ActorSystem instances in the
     // same JVM, here we do it to easily demonstrate these ActorSytems (which would be in separate JVM's)
     // talking to each other.
-    ports foreach { port =>
+
+    ports.foreach { port =>
       // Override the configuration of the port
-      val config = ConfigFactory
-        .parseString("akka.remote.artery.canonical.port=" + port)
-        .withFallback(ConfigFactory.load())
-
-      // Create an Akka system
-      val system = ActorSystem("ShardingSystem", config)
-      // Create an actor that starts the sharding and sends random messages
-      system.actorOf(Devices.props())
+      val config =
+        ConfigFactory.parseString("akka.remote.artery.canonical.port=" + port).withFallback(ConfigFactory.load())
+
+      // Create an Akka system and an actor that starts the sharding
+      // and sends random messages
+      val system = ActorSystem[Message](TemperatureService(), "ShardingSystem", config)
+
+      val cluster = Cluster(system)
+      cluster.manager ! Join(cluster.selfMember.address)
```

Comment:
remove from here, already done in TemperatureService, and such initialization should be done in the root guardian instead of outside

Reply:
Er, forgot I added there and to remove in the app, thanks. Oddly I didn't cluster join in the original sample.